rust/core-lib/src/tabs.rs

```diff
@@ -371,9 +372,13 @@ impl<W: Write + Send + 'static> Documents<W> {
             let buffer_id = self.next_buffer_id();
             self.new_empty_view(rpc_peer, &view_id, buffer_id);
         }
+
+        // schedule plugin notification for after the view has received an identifer
+        let view_id2 = view_id.to_owned();
         let init_info = self.buffers.lock().editor_for_view(&view_id)
             .unwrap().plugin_init_info();
-        self.plugins.document_new(&view_id, init_info);
+        self.plugins.run_on_idle(move |p| p.document_new(&view_id2, &init_info));
```

Comment:
@trishume I pushed some changes addressing a bunch of your comments; the last commit (cd0b001) _slightly_ cleans up the `PluginManagerRef` stuff. However, if there's a way to avoid that mess altogether I'm happy to get rid of it.

One option I considered initially was a much less general solution, where we just stash the `view_id` and  the `init_info`, and send `document_new` directly if they've been stashed. 

Also: a bunch of the ugliness of this stuff is because going through the lock unnecessarily on every pass of the run loop felt bad to me, but maybe that's a grossly premature consideration? My intuition on some of these systems performance topics isn't very well calibrated.